### ISSUE TITLE
Normalizations in architectures

### ISSUE THREAD

**Issue by DrewGalbraith** [DRAGNLabs/301r_retnet#93]
I see RMSNorm being [used in RetNet](https://github.com/DRAGNLabs/301r_retnet/blob/main/src/torchscale/architecture/retnet.py#L95), LayerNorm [in the Transformer](https://github.com/DRAGNLabs/301r_retnet/blob/main/src/torchscale/architecture/decoder.py#L48), no apparent normalization in [LongNet](https://github.com/DRAGNLabs/301r_retnet/blob/main/src/torchscale/architecture/longnet.py), and we're considering not doing a gridsearch over weight decay because apparently if you have a LR scheduler (or adaptive LR like we do with ADAM) combined with batchnorm, weight decay becomes redundant. Who said anything about batchnorm? Well, [we did](https://github.com/DRAGNLabs/301r_retnet/blob/main/src/train_model.py#L176), but I can't figure out if we actually use it anywhere. And it doesn't make much sense in an NLP task, as batchnorm seems to typically apply to CV tasks. <sup>[1](https://stats.stackexchange.com/questions/474440/why-do-transformers-use-layer-norm-instead-of-batch-norm)</sup>

RMSNorm seems to be a direct substitution for LayerNorm<sup>[2](https://dl.acm.org/doi/pdf/10.5555/3454287.3455397)</sup>, but a little better, so I think we should center on that, but it depends what normalization LongNet lends itself to if any. _We should search the LongNet paper for details_ on this. We should also see some TB logs for retnet and transformer using RMS and LayerNorm to:
- Make sure both architectures function under both normalization techniques
- See which normalization performs better for which/both architecutres.

**Comment by DrewGalbraith** [comment]
Ok, first off, the only publishing of the [LongNet paper](https://arxiv.org/pdf/2307.02486) is on arXiv, same as RetNet, which doesn't look good for our paper. Anyway, searching the following terms yielded no information about it:
`scaling, unitization, standard*, norm*, regular*, stable*, smooth*`

**Comment by DrewGalbraith** [comment]
Found it on the GitHub, actually. LongNet [inherits](https://github.com/microsoft/torchscale/blob/main/torchscale/model/LongNet.py#L4-L10) Transformer's decoder classes with use layernorm.

So we should make sure both RMS and Layer norms work for all 3 architectures.

**Comment by DrewGalbraith** [comment]
For tensorboard outputs with each architecture using both normalizations, see #80. However, see the following correction:

I forgot to turn off `RMSNorm` in `multiscale_retention.py`. I reran RetNet with this and added it into the RetNet logs. It is very comparablet to using `RMS norm` in `multiscale_retnention.py` and `layer norm` in `models.py `. The grey line, `retnet_layer_only`, represents the corrected run:
<img width="771" alt="image" src="https://github.com/user-attachments/assets/caee18b2-92a9-4f79-b315-a1ab2484652a" />

**Comment by DrewGalbraith** [comment]
In the [first graph][(https://github.com/DRAGNLabs/301r_retnet/issues/80#issuecomment-2573637657), a curve titled `transformer_layernorm_nosync` directly overlaps `transformer_layernorm` and `transformer_rms`. This was the result of commenting out the `sync_batchnorm` [command in train_model.py](https://github.com/DRAGNLabs/301r_retnet/blob/main/src/train_model.py#L176) during a training run with 8 GPUs. My investigation of the code showed we weren't using batchnorm anywhere, and this confirmed it. Recommending we remove this line for clarity.
